milton-processor/cache.ts

```diff
@@ -49,42 +52,67 @@ function sha256Hash(url: string): string {
   return base64url.default.fromBase64(crypto.createHash('sha256').update(url).digest('base64'));
 }
 
-function urlToKey(url: string): string {
-  return `${CONFIG.cache_prefix}/${sha256Hash(url)}.json`;
+function urlToFolder(url: string): string {
+  const folderPath =  `${CONFIG.cache_prefix}/${sha256Hash(url)}`;
+  console.debug(`URL ${url} located at: ${folderPath}`);
+  return folderPath;
 }
 
-function urlToFile(url: string): gcs.File {
+function urlToFile(url: string, filename: string): gcs.File {
   const storage = createStorageClient();
   const bucket = storage.bucket('scribe-storage');
-  const key = urlToKey(url);
-  console.debug(`URL ${url} located at: ${key}`);
+  const key = `${urlToFolder(url)}/${filename}`;
   return bucket.file(key);
 }
 
-export function save(url: string, contents: Manuscript) {
-  const file = urlToFile(url);
+export function saveRawContents(url: string, rawContents: any, contentType?: string) {
+  const file = urlToFile(url, RAW_CONTENTS_PATH);
+  file.save(rawContents).then(() => {
+    if (contentType) {
+      file.setMetadata({contentType});
+    }
+    console.log(`Saved raw contents of URL (${url}) to cache`);
+  }).catch((err) => {
+    console.error(`Failed to save raw contents of URL (${url}) with error: ${err.message}`);
+  });
+}
+
+export function fetchRawContents(url: string): Promise<string> {
+  const file = urlToFile(url, RAW_CONTENTS_PATH);
+  const cachedContents = file.download().then((data) => {
+    const fileContents = data[0].toString();
+    return fileContents;
```

Comment:
You can elide the variable here (and make this lambda into a single expression)

Reply:
updated